python/lsst/jointcal/jointcal.py

```diff
@@ -174,6 +174,42 @@ def lookupStaticCalibrations(datasetType, registry, quantumDataId, collections):
                                   findFirst=True)
 
 
+def lookupVisitRefCats(datasetType, registry, quantumDataId, collections):
+    """Lookup function that finds all refcats for all visits that overlap a
+    tract, rather than just the refcats that directly overlap the tract.
+
+    Parameters
+    ----------
+    datasetType : `lsst.daf.butler.DatasetType`
+        Type of dataset being searched for.
+    registry : `lsst.daf.butler.Registry`
+        Data repository registry to search.
+    quantumDataId : `lsst.daf.butler.DataCoordinate`
+        Data ID of the quantum; expected to be something we can use as a
+        constraint to query for overlapping visits.
+    collections : `Iterable` [ `str` ]
+        Collections that should be searched.
+
+    Returns
+    -------
+    refs : `Iterator` [ `lsst.daf.butler.DatasetRef` ]
+        Iterator over refcat references.
+    """
+    refs = set()
+    # Use .expanded() because we need data IDs with regions (in both the outer
+    # loop and the inner loop).
```

Comment:
Which are the "inner" and "outer" loop here? Do you mean something like "in the dataIds to iterate over and the result to update the datasetRefs with"?